controller/controller.go

```diff
@@ -163,7 +163,21 @@ func (c *Controller) DeleteFunc(obj interface{}) {
 	resourceSet, err := c.resourceRouter.ResourceSet(obj)
 	if IsNoResourceSet(err) {
 		// In case the resource router is not able to find any resource set to
-		// handle the reconciled runtime object, we stop here.
+		// handle the reconciled runtime object, we stop here. Note that we just
+		// remove the finalizer regardless because at this point there will never be
+		// a chance to remove it otherwhise because nobody wanted to handle this
+		// runtime object anyway.
+
+		c.logger.Log("level", "debug", "message", "removing finalizer from runtime object")
+
+		err = c.removeFinalizer(obj)
```

Comment:
I did not see a reason why the context is there. It is not used for anything. It is also that for adding the finalizer there is no context taken. If we need it we can add it back again. Just want to get it straight now.